test/remote/gateways/remote_migs_test.rb

```diff
@@ -104,6 +104,16 @@ def test_refund
     # assert_equal 'Approved', response.message
   end
 
+  def test_passes_tx_source
+    # Purchase returns a successful response when a valid tx_source parameter is sent.
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(tx_source: 'MOTO'))
```

Comment:
As I'm going through and writing more remote tests, I was looking more closely at these lines. Why is this written as an `assert` statement? Should this `assert` be removed?

Reply:
This is a fairly common pattern, but I haven't really examined why. It would give us more "resolution" for failures, since checking that this step fails is distinct from the logic of the `assert_success` check.

Reply:
Interesting; thanks for the insight. I'll leave these in for more test granularity.